File: client/python/examples/usage.py
import sys
from dkv import DKVClient, insecure_credentials


def main():
    print("Connecting to DKV server...")
    client = DKVClient.connect("127.0.0.1:50051", insecure_credentials())


    key = "python-example-key"
    val = b"Hello DKV from Python!"

    try:
        print("Storing value under key...")
        client.set(key, val)

        print("Retrieving value...")
        retrieved_val = client.get(key)
        if retrieved_val is not None:
            string_val = retrieved_val.decode("utf-8")
            print(f"Success! Retrieved value: \"{string_val}\"")
        else:
            print("Key not found!")

        print("Deleting key...")
        client.delete(key)

        print("Verifying deletion...")
        check = client.get(key)
        print(f"Key exists: {check is not None}")

        print("Done!")
    except Exception as e:
        print(f"An error occurred: {e}", file=sys.stderr)
    finally:
        client.close()

if __name__ == "__main__":
    main()

File: client/rust/examples/usage.rs
use oryx::OryxClient;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("Connecting to ORYX server...");
    let client = OryxClient::connect("http://127.0.0.1:50051").await?;

    let key = "rust-example-key";
    let val = b"Hello ORYX from robust Async Rust!".to_vec();

    println!("Storing value under key...");
    client.set(key, val.clone()).await?;

    println!("Retrieving value...");
    if let Some(retrieved_val) = client.get(key).await? {
        let string_val = String::from_utf8(retrieved_val)?;
        println!("Success! Retrieved value: \"{}\"", string_val);
    } else {
        println!("Key not found!");
    }

    println!("Deleting key...");
    let existed = client.delete(key).await?;
    println!("Deleted key (existed: {})", existed);

    println!("Verifying deletion...");
    let check = client.get(key).await?;
    println!("Key exists: {}", check.is_some());

    println!("Done!");
    Ok(())
}
